keras/saving/pickle_utils_test.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for pickling / deepcopying of Keras Models."""
+
+import tensorflow.compat.v2 as tf
+
+import copy
+import pickle
+
+import numpy as np
+
+from keras import keras_parameterized
+from keras import layers as layers_module
+from keras.engine import input_layer
+from keras.engine import training as training_module
+
+
+class TestPickleProtocol(keras_parameterized.TestCase):
+  """Tests pickle protoocol support.
+  """
+
+  @keras_parameterized.run_all_keras_modes
+  def test_pickle_model(self):
+    """Test copy.copy, copy.deepcopy and pickle on Functional Model."""
```

Comment:
Let's test all model types (Sequential, Functional, subclass). We have a parameterization for that. `@keras_parameterized.run_with_all_model_types`. See examples.

Reply:
I attempted to incorporate this parametrization based on other examples, but I'm not 100% sure I got it right.

Reply:
It seems to be working, I'm seeing tests run for `sequential`, `subclass`, etc.